Port bind skipped at server startup

**Describe the bug**  
In an extremely rare scenario, configured ports can silently fail to bind - preventing Liberty from using them. This issue is caused by a subtle race condition in the code responsible for delaying the port bind until the server is ready to handle traffic.

In the failing scenario, the _port started_ message will not be emitted - for example the following message will be missing:
```
CWWKO0219I: TCP Channel defaultHttpEndpoint has been started and is now listening for requests on host *  (IPv4) port 9080.
```
and the following FFDC will be seen:
```
Exception = java.lang.RuntimeException
Source = com.ibm.ws.tcpchannel.internal.NBAcceptChannelSelector
probeid = 254
Stack Dump = java.lang.RuntimeException: java.nio.channels.NotYetBoundException
	at com.ibm.ws.tcpchannel.internal.NBAcceptChannelSelector.performRequest(NBAcceptChannelSelector.java:358)
	at com.ibm.ws.tcpchannel.internal.ChannelSelector.run(ChannelSelector.java:257)
	at java.base/java.lang.Thread.run(Thread.java:866)
Caused by: java.nio.channels.NotYetBoundException
	at java.base/sun.nio.ch.ServerSocketChannelImpl.begin(ServerSocketChannelImpl.java:246)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.accept(ServerSocketChannelImpl.java:283)
	at com.ibm.ws.tcpchannel.internal.NBAcceptChannelSelector.performRequest(NBAcceptChannelSelector.java:278)
	... 2 more
```
**Steps to Reproduce**
Configure one or more ports to start, for example: 
```
    <httpEndpoint id="defaultHttpEndpoint"
                  host="*"
                  httpPort="9080"
                  httpsPort="9443">
    </httpEndpoint>   
```
And start the server. In this scenario, one or more of the ports will silently fail to bind.

**Expected behavior**  
All configured ports should start, or if there is a problem some meaningful error message should be logged.

**Diagnostic information:**  
 - OpenLiberty Version: All
 - Affected feature(s) All
 - Java Version: reported on `IBM Semeru Runtime Open Edition (11.0.12+7)`
 - server.xml configuration: N/A, just need multiple ports

--- a/dev/com.ibm.ws.channelfw/src/com/ibm/websphere/channelfw/osgi/CHFWBundle.java
+++ b/dev/com.ibm.ws.channelfw/src/com/ibm/websphere/channelfw/osgi/CHFWBundle.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2009, 2020 IBM Corporation and others.
+ * Copyright (c) 2009, 2021 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at
@@ -95,8 +95,9 @@ public class CHFWBundle implements ServerQuiesceListener {
     /** Reference to the executor service -- required */
     private ExecutorService executorService = null;
 
-    private static AtomicBoolean serverCompletelyStarted = new AtomicBoolean(false);
-    private static Queue<Callable<?>> serverStartedTasks = new LinkedBlockingQueue<>();
+    private static final AtomicBoolean serverCompletelyStarted = new AtomicBoolean(false);
+    private static final AtomicBoolean useServerStartingTaskQueue = new AtomicBoolean(true);
+    private static final Queue<Callable<?>> serverStartedTasks = new LinkedBlockingQueue<>();
     private static Object syncStarted = new Object() {
     }; // use brackets/inner class to make lock appear in dumps using class name
 
@@ -242,11 +243,11 @@ public class CHFWBundle implements ServerQuiesceListener {
         // set will be called when the ServerStarted service has been registered (by the FeatureManager as of 9/2015).  This is a signal that
         // the server is fully started, but before the "smarter planet" message has been output. Use this signal to run tasks, mostly likely tasks that will
         // finish the port listening logic, that need to run at the end of server startup
-
         Callable<?> task;
         if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
             Tr.debug(this, tc, "CHFW signaled- Server Completely Started signal received");
         }
+        useServerStartingTaskQueue.set(false);
         while ((task = serverStartedTasks.poll()) != null) {
             try {
                 task.call();
@@ -256,7 +257,6 @@ public class CHFWBundle implements ServerQuiesceListener {
                 }
             }
         }
-
         synchronized (syncStarted) {
             serverCompletelyStarted.set(true);
             syncStarted.notifyAll();
@@ -273,7 +273,7 @@ public class CHFWBundle implements ServerQuiesceListener {
      */
     public static <T> T runWhenServerStarted(Callable<T> callable) throws Exception {
         synchronized (syncStarted) {
-            if (!serverCompletelyStarted.get()) {
+            if (useServerStartingTaskQueue.get()) {
                 serverStartedTasks.add(callable);
                 return null;
             }

